qoajad_backend/src/main/java/com/qoajad/backend/controller/AppointmentController.java

```diff
@@ -26,38 +27,80 @@ public AppointmentController(@Qualifier("defaultAppointmentService") final Appoi
         this.appointmentService = appointmentService;
     }
 
-    @RequestMapping(value = "/appointment/find/{id}", method = RequestMethod.GET)
-    public ResponseEntity<Appointment> findAppointment(@PathVariable("id") final int id) {
-        ResponseEntity<Appointment> response;
+    @RequestMapping(value = "/appointment/all_available_appointments/{hpiName}/{specialtyName}", method = RequestMethod.GET)
+    public ResponseEntity<List<ConsultingRoom>> findAvailableAppointments(@PathVariable String hpiName,
+                                                                          @PathVariable String specialtyName) {
+        ResponseEntity<List<ConsultingRoom>> response;
         try {
-            response = new ResponseEntity<>(appointmentService.findAppointment(id), HttpStatus.OK);
+            response = new ResponseEntity<>(
+                    appointmentService.findAvailableAppointments(hpiName, specialtyName),
+                    HttpStatus.OK);
         } catch (EmptyResultDataAccessException e) {
             response = new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        } catch (DataIntegrityViolationException e) {
+            response = new ResponseEntity<>(HttpStatus.CONFLICT);
+        }
+        return response;
+    }
+
+    @RequestMapping(value = "/appointment/all_user_appointments/{userDocument}", method = RequestMethod.GET)
+    public ResponseEntity<List<Appointment>> findUserAppointments(@PathVariable int userDocument) {
+        ResponseEntity<List<Appointment>> response;
+        try {
+            response = new ResponseEntity<>(
+                    appointmentService.findUserAppointments(userDocument),
+                    HttpStatus.OK);
+        } catch (EmptyResultDataAccessException e) {
+            response = new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        } catch (DataIntegrityViolationException e) {
+            response = new ResponseEntity<>(HttpStatus.CONFLICT);
         }
         return response;
     }
 
     @RequestMapping(value = "/appointment/create", method = RequestMethod.POST)
-    public ResponseEntity<String> createAppointment(@RequestBody CreateAppointment createAppointment, final HttpServletRequest request) {
+    public ResponseEntity<String> createAppointment(@RequestBody CreateAppointment createAppointment) {
         ResponseEntity<String> response;
         try {
-            appointmentService.createAppointment(createAppointment);
+            appointmentService.attemptToCreateAppointment(createAppointment);
```

Comment:
That method shows intention that creating an appointment may or may not be possible but your response is always successful.